source/fab/build_config.py

```diff
@@ -41,36 +42,50 @@ class BuildConfig():
     """
     def __init__(self, project_label: str,
                  tool_box: ToolBox,
-                 multiprocessing: bool = True, n_procs: Optional[int] = None,
+                 mpi: bool,
+                 openmp: bool,
+                 multiprocessing: bool = True,
+                 n_procs: Optional[int] = None,
                  reuse_artefacts: bool = False,
-                 fab_workspace: Optional[Path] = None, two_stage=False,
-                 verbose=False):
+                 fab_workspace: Optional[Path] = None,
+                 two_stage: bool = False,
+                 verbose: bool = False):
```

Comment:
Well, that seems unavoidable - in a complex build systems there are a lot of things to be defined :) Now that mpi/openmp are optional, it might be better? Also, this mostly will be handled by the command line tool in the end, which should have an appropriate man page/help command.